db/apply.go

```
@@ -234,6 +246,67 @@ func (chain *Chain) ApplyFAT0Tx(ei int64, e factom.Entry) (tx *fat0.Transaction,
 	return
 }
 
+var tmp grader.BlockGrader
+
+func (chain *Chain) ApplyFAT2OPR(ei int64, e factom.Entry, eb factom.EBlock, pos int) (err error) {
+	if !eb.IsPopulated() || pos < 0 { // 'processing' entry
```

This really seems like it should return an error. 

Also I believe it is a reasonable expectation that the caller has populated the EBlock. I think it would be best to not check if the eblock is populated and just let panics happen below here as this represents a serious integrity error with the code.

---

both "pending" entries and regular entries call the Apply function, but in the case of pending entries the entryblock passed into the function is `nil` and the position is -1 (unknown):

https://github.com/Factom-Asset-Tokens/fatd/blob/6d658860a2f18dda8d393357da24a9501af7b81f/engine/process.go#L151

it's not an error because at the time it is called, the entry cannot be validated. it can only be validated in the context of an eblock